src/main/java/pl/javastart/restoffers/categories/CategoriesService.java

```diff
@@ -0,0 +1,44 @@
+package pl.javastart.restoffers.categories;
+
+import org.springframework.stereotype.Service;
+import pl.javastart.restoffers.offers.OffersService;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class CategoriesService {
+
+    private final CategoriesRepository categoriesRepository;
+    private final OffersService offersService;
+
+    CategoriesService(CategoriesRepository categoriesRepository, OffersService offersService) {
+        this.categoriesRepository = categoriesRepository;
+        this.offersService = offersService;
+    }
+
+    List<CategoryDto> findAll() {
+        List<CategoryDto> categories = new ArrayList<>();
+        for (Category category : categoriesRepository.findAll()) {
+            CategoryDto categoryDto = new CategoryDto();
+            categoryDto.setId(category.getId());
+            categoryDto.setName(category.getName());
+            categoryDto.setDescription(category.getDescription());
+            categoryDto.setOffers(offersService.countOffersInCategory(category.getName()));
```

Comment:
zamiast wykonywać zapytanie dla każdej kategorii, powinno wystarczyć `category.getOffers().size()`

Reply:
Taka metoda, niestety nie zadziałała. Nie wiem skąd JVM brał sobie tą liczbę, ale przy deklarowaniu tej wartości jako category.getOffers().size(), wyświetlane było cały czas 1. Jakoby dana kategoria miałą tylko jedną ofetrę. Testowałem to pole jako category.getOffers().toString() i w takim przypadku wyświetlane były po jednej ofercie dla jednej kategorii. Pewnie to gdzieś mój błąd ale... taki znalazłem sposób, żeby to ominąć.